### ISSUE TITLE
Problems with plt.AxisAuto() and negative numbers 

### ISSUE THREAD

**Issue by CarloToso** [ScottPlot/ScottPlot#1855]
## Bug Report

**Issue:**  Data points with negative values go out of bounds with AxisAuto().

![test](https://user-images.githubusercontent.com/105941898/169590744-70fa2bd9-3412-4907-affd-18af30075626.jpeg)

```cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Text;

namespace App
{
    public class Program
    {
        public static void Main()
        {
        double[] dataX = new double[] { -1000, -2000, 5000, -400, 200 };
        double[] dataY = new double[] { 1, 4, 9, 16, 25 };
        var plt = new ScottPlot.Plot(1280, 960);
        plt.AddLollipop(dataX, dataY);
        plt.SaveFig("test.jpeg");
        }
    }
}
```

### System Details

* **ScottPlot Version:** 4.1.45
* **Operating System:** Windows 10
* **Application Type:** console
* **.NET Version:** NET Framework 4.61, Net5.0

**Comment by swharden** [comment]
Thanks for reporting this @CarloToso! This issue may be restricted to just lollipop plots (and perhaps Cleveland plots)

The best fix may be to implement a new `GetAxisLimits()` inside `LollipopPlot` (which currently inherits from and uses the data limit detection logic in `BarPlotBase` which is way more complicated than it needs to be)

https://github.com/ScottPlot/ScottPlot/blob/d3b9c13b67d0344cf68e6e1cb7893fc0f1785e7f/src/ScottPlot4/ScottPlot/Plottable/LollipopPlot.cs#L16

https://github.com/ScottPlot/ScottPlot/blob/d3b9c13b67d0344cf68e6e1cb7893fc0f1785e7f/src/ScottPlot4/ScottPlot/Plottable/BarPlotBase.cs#L108-L138

**Comment by CarloToso** [comment]
**Plot types that seem to have this problem:**
Lollipop, Bar, Crosshair (when you use it alone it doesn't center the plot on the intersection)

**Plot types that don't seem to have this problem:**
BubblePlot, Scatter, ScatterLines, ScatterPoints, Cleveland, Fill

**Comment by bclehmann** [comment]
I created a PR #1857, but it only touches bar plots (and thus lollipop, as they share a base class).

I didn't touch Crosshairs because judging by the code it seems it is explicitly asking not to be part of axis calculation: https://github.com/ScottPlot/ScottPlot/blob/main/src/ScottPlot4/ScottPlot/Plottable/Crosshair.cs#L100
```cs
public AxisLimits GetAxisLimits() => new(double.NaN, double.NaN, double.NaN, double.NaN);
```

I would lean towards crosshairs being part of axis calculation, but I would defer to Scott in case there's a specific reason they are excluded. Notably `VLine` and `HLine` are included by default, but can be excluded through an `IgnoreAxisAuto` property. To me, VLine and HLine serve a similar purpose as a crosshair, so that might be a good thing to replicate.

**Comment by swharden** [comment]
I'll update Crosshair to behave more like the vertical/horizontal lines. It seems like it should report the center of the crosshair, but also have the option to `IgnoreAxisAuto`.

It took me a minute to figure out why the `IgnoreAxisAuto` exists, but I found it: Consider situations where axis lines (or a crosshair) follow the mouse continuously. If you zoom really far out and middle-click (to call AxisAuto()) it doesn't zoom back in properly because the mouse-following objects are included in the limit detection.

**Comment by swharden** [comment]
Actually, I won't do this now, but it seems like there should be interfaces like `IHasAxisLimits` rather than using a `bool` to selectively disable this behavior. I'll loop back around after this PR is merged and see about implementing that in #1855
